src/core/sim.py

```diff
@@ -72,9 +91,32 @@ def should_stop(self) -> bool:
             log.error("Stopping sim because it's running too long")
             return True
 
-        if (state.x**2 + state.y**2 + state.z**2)**0.5 < R_EARTH:
+        if (state.x**2 + state.y**2 + state.z**2) ** 0.5 < R_EARTH:
             log.error("Stopping sim because craft is inside the Earth")
             log.debug(f"r={(state.x**2 + state.y**2 + state.z**2)**0.5} < {R_EARTH}")
+            data_plot = Plot(self.xlocs, self.ylocs, self.zlocs, self.state)
+            data_plot.plot_data()
```

Comment:
Plotting should not occur within sim.py, only externally to keep this as modular as possible. (i.e. I'd like to be able to plot all trajectories, not just ones that intercept the earth)